src/dataflow/source/kafka.rs

```diff
@@ -63,6 +64,45 @@ impl<'a> From<&BorrowedMessage<'a>> for MessageParts {
     }
 }
 
+// Refresh our metadata from Kafka, which is necessary in order to start receiving messages
+// for new partitions. Return whether the check succeeded and returned expectedly many partitions.
+fn refresh_kafka_metadata(
+    consumer: &mut BaseConsumer<GlueConsumerContext>,
+    expected_partitions: usize,
+    topic: &str,
+    activator: &Activator,
+) -> bool {
+    let result = match consumer.fetch_metadata(Some(topic), Duration::from_secs(1)) {
+        Ok(md) => match md.topics().iter().find(|mdt| mdt.name() == topic) {
+            None => {
+                warn!("Topic {} not found in Kafka metadata", topic);
+                false
+            }
+            Some(mdt) => {
+                let partitions = mdt.partitions().len();
+                if partitions < expected_partitions {
+                    warn!(
+                        "Topic {} does not have as many partitions as expected ({} < {})",
+                        topic, partitions, expected_partitions
+                    );
+                    false
+                } else {
+                    true
+                }
+            }
+        },
+        Err(e) => {
+            warn!("Error refreshing Kafka metadata: {}", e);
+            false
+        }
+    };
+    if !result {
+        warn!("Descheduling for 1s to wait for Kafka metadata to catch up");
```

Comment:
I'm not entirely sure this is a good idea. Partitions + topic get created asynchronously but relatively quickly. I can understand the benefit of not looping and instead calling .activate(), but I don't  think we should wait (or at least not as long), as I think this is likely going to increase latency noticeably on startup.